torchvision/transforms/transforms.py

```diff
@@ -528,6 +529,61 @@ def __repr__(self):
         return self.__class__.__name__ + '(p={})'.format(self.p)
 
 
+class RandomPerspective(object):
+    """Performs Perspective transformation of the given PIL Image randomly with a given probability.
+
+    Args:
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        interpolation : Default- Image.BICUBIC
+    """
+
+    def __init__(self, interpolation=Image.BICUBIC, p=0.5):
+        self.p = p
+        self.interpolation = interpolation
+
+    def __call__(self, img):
+        """
+        Args:
+            img (PIL Image): Image to be Perspectively transformed.
+
+        Returns:
+            PIL Image: Random perspectivley transformed image.
+        """
+        if not F._is_pil_image(img):
+            raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
+
+        if random.random() < self.p:
+            width, height = img.size
+            startpoints, endpoints = self.get_params(width, height)
+            return F.perspective(img, startpoints, endpoints, self.interpolation)
+        return img
+
+    @staticmethod
+    def get_params(width, height):
+        """Get parameters for ``perspective`` for a random perspective transform.
+
+        Args:
+            width : width of the image.
+            height : height of the image.
+
+        Returns:
+            List containing [top-left, top-right, bottom-right, bottom-left] of the orignal image,
+            List containing [top-left, top-right, bottom-right, bottom-left] of the transformed image.
+        """
+        half_height = int(height / 2)
```

Comment:
I think those two values could be kind of given as input to the transform.
Right now the user doesn't have any control on the amount of transformation that is applied, and the image can be pretty drastically distorted. It might be good to give some control to the user for that.

A possibility would be to let the user specify a factor of distortion. A small factor would make for consistently small random distortions, while a large factor can sample distortions up to `height / 2`.

So, here is one proposal (but let me know what you think):

```python
max_height = int(height / 2 * scale_height)  # need better name
max_width = int(width / 2 * scale_width)
topleft = (random.randint(0, max_width), random.randint(0, max_height))
topright = (random.randint(width - max_width, width - 1), random.randint(0, max_height))
...
```

What do you think?

Reply:
@fmassa , i think adding one parameter would suffice 
This means.
If i have a small alpha value , then the new coordinates would be similiar to the orignal ones or else just try and increase the distortion randomly.

`        half_height = int(height / 2) `
       ` half_width = int(width / 2) `
        `topleft = (random.randint(0, alpha*half_width), random.randint(0, alpha*half_height))`
       ` topright = (random.randint(width - alpha*half_width - 1, width - 1), random.randint(0, alpha*half_height))`
        `botright = (random.randint(width - alpha*half_width - 1, width - 1), random.randint(height - alpha*half_height - 1, height - 1)) `
        `botleft = (random.randint(0, alpha*half_width), random.randint(height - alpha*half_height - 1, height - 1))`

What are your thoughts on this ?

Reply:
Having one parameter sounds good to me as well. It could be named something like `distortion_scale` or something like that.

Reply:
@fmassa , just made changes could you review them .